src/bindgen/ir/ty.rs

```diff
@@ -1002,6 +1005,32 @@ impl Type {
             Type::FuncPtr { .. } => true,
         }
     }
+
+    // Search and replace specific associated types with concrete types
+    pub fn resolve_assoc_types(&mut self, resolver: &AssocTypeResolver) {
+        match self {
+            Type::Ptr { ty: ty_, .. } => {
+                ty_.resolve_assoc_types(resolver);
+            }
+            Type::Array(ty_, _) => {
+                ty_.resolve_assoc_types(resolver);
+            }
+            Type::FuncPtr { ret, args, .. } => {
+                ret.resolve_assoc_types(resolver);
+                for (_, ty_) in args {
+                    ty_.resolve_assoc_types(resolver);
+                }
+            }
+            Type::Path(generic_path) => {
+                if let Some(id) = generic_path.assoc() {
+                    if let Some(concrete) = resolver.get(id) {
+                        *self = concrete.clone();
+                    }
+                }
+            }
+            _ => {}
```

Comment:
```suggestion
            Type::Primitive(_) => {}
```
don't use wildcards, they aren't future proof